### ISSUE TITLE
Binary character in header value cause lua error, 500 response status code when using request-transformer plugin

### ISSUE THREAD

**Issue by cr8zi** [Kong/kong#6264]
### Summary

If request-transformer plugin is enabled and request header value contains binary character, kong return 500 response status code, lua error in log. We use kong ingress controller for kubernetes, but root cause is in kong proxy, request-transformer plugin.

### Steps To Reproduce

1. Install Kong 2.0 or 2.1
2. Use simple dbless config:
```
_format_version: "1.1"

services:
- name: my-service
  url: http://api-service
  plugins:
  - name: request-transformer
    config: 
      remove:
        headers:
          - X-Test-Header
  routes:
  - name: my-route
    paths:
    - /v1/test
```
3. curl example:
```
curl -v http://api.example.com:8000/v1/test -H "`echo -e "Accept: application/json\x1f"`" --resolve api.example.com:8000:127.0.0.1
* Added api.example.com:8000:127.0.0.1 to DNS cache
* Hostname api.example.com was found in DNS cache
*   Trying 127.0.0.1:8000...
* TCP_NODELAY set
* Connected to api.example.com (127.0.0.1) port 8000 (#0)
> GET /v1/test HTTP/1.1
> Host: api.example.com:8000
> User-Agent: curl/7.68.0
> Accept: application/json
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Internal Server Error
< Date: Fri, 21 Aug 2020 08:21:31 GMT
< Content-Type: application/json; charset=utf-8
< Connection: keep-alive
< Content-Length: 42
< X-Kong-Response-Latency: 2
< Server: kong/2.1.3
< 
* Connection #0 to host api.example.com left intact
{"message":"An unexpected error occurred"}
```
4. kong error log:
```
2020/08/21 09:58:24 [error] 29#0: *153 unsafe byte "0x1f" in header "application/json\x1F", client: 172.23.0.1, server: kong, request: "GET /v1/test HTTP/1.1", host: "api.example.com:8000"
2020/08/21 09:58:24 [error] 29#0: *153 lua coroutine: runtime error: /usr/local/openresty/lualib/resty/core/request.lua:335: error
stack traceback:
coroutine 0:
[C]: in function 'error'
 	/usr/local/openresty/lualib/resty/core/request.lua:335: in function 'set_header'
 	/usr/local/share/lua/5.1/kong/pdk/service/request.lua:400: in function 'set_headers'
 	...hare/lua/5.1/kong/plugins/request-transformer/access.lua:229: in function 'transform_headers'
 	...hare/lua/5.1/kong/plugins/request-transformer/access.lua:512: in function 'execute'
 	...are/lua/5.1/kong/plugins/request-transformer/handler.lua:11: in function <...are/lua/5.1/kong/plugins/request-transformer/handler.lua:10>
coroutine 1:
 	[C]: in function 'resume'
 	coroutine.wrap:21: in function <coroutine.wrap:21>
 	/usr/local/share/lua/5.1/kong/init.lua:757: in function 'access'
 	access_by_lua(nginx-kong.conf:91):2: in main chunk, client: 172.23.0.1, server: kong, request: "GET /v1/test HTTP/1.1", host: "api.example.com:8000"
2020/08/21 09:58:24 [error] 29#0: *153 [kong] init.lua:759 [request-transformer] /usr/local/openresty/lualib/resty/core/request.lua:335: error, client: 172.23.0.1, server: kong, request: "GET /v1/test HTTP/1.1", host: "api.example.com:8000"
```
### Additional Details & Logs

- Kong version 2.0,2.1

**Comment by hutchic** [comment]
I'm curious what the context or your use case is such that a binary character exuists in your header?

**Comment by hishamhm** [comment]
```
runtime error: /usr/local/openresty/lualib/resty/core/request.lua:335: error
```

at first glance it seems to have failed inside OpenResty?

**Comment by cr8zi** [comment]
> I'm curious what the context or your use case is such that a binary character exuists in your header?

Got it from our QA, I think they used some generated test for API. I can image that it can cause some performance/log flood problems if you terminate TLS directly in Kong, without anything validating request headers before Kong.

We use Kong ingress controller for Kubernetes. What is interesting, if we use the request-transformer plugin on route (mainly rewriting path in request), request fails with http response status 500; but if request-transformer plugin is not enabled on route, Kong return http response status 400 Bad Request: invalid header value.

**Comment by bungle** [comment]
This seems to be fixed in `OpenResty 1.17.x.x`. At least I could not reproduce it anymore. @cr8zi can you try out `2.2.0-alpha`? I will close this as I think it is fixed. @cr8zi, please reopen this if you still see this with `2.2`.
